unison-share-api/src/Unison/Server/CodebaseServer.hs

```diff
@@ -285,31 +300,52 @@ serveIndex path = do
                   <> " environment variable to the directory where the UI is installed."
           }
 
-serveUI :: Handler () -> FilePath -> Server WebUI
-serveUI tryAuth path _ = tryAuth *> serveIndex path
+serveUI :: FilePath -> Server WebUI
+serveUI path _ = serveIndex path
 
 server ::
   Rt.Runtime Symbol ->
   Codebase IO Symbol Ann ->
   FilePath ->
   Strict.ByteString ->
-  Server AuthedServerAPI
-server rt codebase uiPath token =
+  Server AppAPI
+server rt codebase uiPath expectedToken =
   serveDirectoryWebApp (uiPath </> "static")
-    :<|> ( \token ->
-             serveUI (tryAuth token) uiPath
-               :<|> unisonApi token
-               :<|> serveOpenAPI
-               :<|> Tagged serveDocs
-         )
+    :<|> hoistWithAuth serverAPI expectedToken serveServer
```

Comment:
Rather than pass the `tryAuth` handler action into every handler, which just runs it before handling the request, I now do it as a form of middleware using `hoistWithAuth`, this splits authorization concerns from the actual action we're performing, which makes it easier to use the handlers in Enlil, since they'll be using a different form of authorization.